docs/documentation/advanced/unkown-fields.fsx

```diff
@@ -0,0 +1,302 @@
+(**
+---
+layout: standard
+title: Unkown fields
+---
+*)
+
+(*** hide ***)
+
+#I "../../../src/bin/Debug/netstandard2.0"
+#r "Thoth.Json.dll"
+
+open Thoth.Json
+
+(**
+
+It can happens that you receive a JSON but you don't know which fields are going to be present.
+
+Consider the following JSON:
+
+```json
+{
+    "ts": "2020-01-01T00:00:00Z",
+    "EUR_PLN": { "rate": "4.55" },
+    "GBP_PLN": { "error": "Rate is not available at the moment" },
+    "USD_PLN": { "rate": "4.01" }
+}
+```
+
+In this example, we know that:
+
+- The `ts` field required
+- There is an unkown number of `Rate` fields which consists of:
+    - The key property which has the format:
+        1. `sourceCurrency`
+        2. `_`
+        3. `targetCurrency`
+    - The value should have a `rate` field which is a `decimal`
+
+We are now going to write a docuder capable of handling such a JSON.
+
+## Helpers
+
+First, we are going to define some helpers to make the code more readable.
+*)
+
+module Option =
+
+    /// Return the value if there is one, otherwise throw an exception
+    let unwrap (value : 'T option) =
+        match value with
+        | Some value -> value
+        | None -> failwith "Option.unwrap: None"
+
+module List =
+
+    /// Returns all the elements in the list which have a value
+    let unwrapOption (list : ('T option) list) =
+        list
+        |> List.filter Option.isSome
+        |> List.map Option.unwrap
+
+(**
+
+## Custom decoders
+
+We need some specific decoder in order to work with our JSON.
+
+*)
+
+module Decode =
+
+(**
+
+1. We need a decoder which ignore failed decoder
+
+*)
+
+    let ignoreFail (decoder : Decoder<'T>) : Decoder<'T option> =
+        fun path token ->
+            match decoder path token with
+            | Ok x -> Ok(Some x)
+            | Error _ -> Ok None
+
+(**
+
+2. We need a decoder which decodes all the object fields and only keep the valid ones
+
+*)
+
+    let keyValueOptions (decoder: Decoder<'a option>) : Decoder<(string * 'a) list> =
+        decoder
+        |> Decode.keyValuePairs
+        |> Decode.map (
+            List.collect
+                (fun (key, aOpt) ->
+                    match aOpt with
+                    | Some a -> [ key, a ]
+                    | None -> [])
+        )
+
+(**
+
+## Define our domain types
```

Comment:
I am not sure about this whole section. We are supposed to explain the reader how to decode a structure and not how to overall organize the code in theirs project. 

For example, in my case all the advises from this section would be counter-productive because I have a common structure like `RateObject` and then many different providers serve many different JSONs and all I want is to fit these many different JSONs into one common `RateObject list` type.

I would rather focus on how to pull the data out of the JSON and not how to organize decoder functions within a `Ts` or `RateObject` modules which would just not work for me and would be distracting.

Reply:
Without the domain logic it is really difficult if not impossible to understand what types we are trying to use.

For example, in your issues I had to re-read it 10-15 times before understanding what was going on. And this is not just because the JSON decoders are complex but also because I didn't have the F# types we are trying to map to.

Defining the domain logic, is a skill to have with Thoth.Json too because you need to understand how to translate the JSON into an F# type.

A note, can be added to say that this is just one of modelling the JSON and users can have their own depending on how they think.

Reply:
Because, I didn't have the F# type logic it was really hard for me to understand what your decoders where doing. 

So, it was me trying to make sense of the JSON and what output was expected. It is possible that I modelled a bit too much the JSON so I will try to rework it a bit.